ando/tools/generator/tests/test_utils.py

```diff
@@ -20,44 +23,45 @@ def test_create_file(self):
         self.assertTrue(Path(path_to_save).exists())
 
     def test_create_dummy_file(self):
-        df = pd.read_csv(test_directory / 'test_files' / 'participants.tsv', sep='\t')
+        df = pd.read_csv(test_directory / 'test_files' / 'participants.tsv',
+                         sep='\t', index_col=0)
         path_to_save = test_directory / "test_create_dummy_file.tsv"
         save_tsv(df, path_to_save)
 
         self.assertTrue(Path(path_to_save).exists())
-        df_test = pd.read_csv(test_directory / "test_create_dummy_file.tsv" , sep='\t')
-        self.assertTrue(df_test.equals(df))
-
-    def test_merge_existing_file(self):
+        df_read = pd.read_csv(test_directory / "test_create_dummy_file.tsv",
+                              sep='\t', index_col=0)
+        self.assertTrue(df_read.equals(df))
 
+    def test_save_to_existing_file(self):
         a = pd.DataFrame({
-            "a": [1],
-            "c": [0],
-            "d": [2]
+            "i": [1, 2, 3, 4],
+            "a": ['a1', 'a2', 'a3', None],
+            "b": [None, 'b2', None, 'b4']
         })
+        a.set_index('i', inplace=True)
         b = pd.DataFrame({
-            "a": [0],
-            "b": [1],
-
```

Comment:
Why those changes ?

Reply:
I made the example a bit more realistic in the sense that there's more than 2 columns. With the new examples we can cover all these cases:
- none of the two dfs have a value for a field
- only one of the two dfs has a value for a field

Also I now ensure that column 0 is always used as index and therefore is also the basis for merging multiple dataframes (see `merge_dfs_by_index` below)